src/worker/main.ts

```diff
@@ -1 +1,66 @@
-self.postMessage("hello, from worker");
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
+
+/**
+ * A "fake" type; it's a string, but its contents MUST be a valid BCP-47 tag.
+ */
+type BCP47 = string & { __bcp46__: true };
+
+/**
+ * A model ID, according to spec.
+ */
+interface ModelID {
+  readonly author: string;
+  readonly language: string;
+  readonly tag: string;
+}
+
+interface CompileMessage {
+  readonly method: "compile";
+  readonly modelID: ModelID;
+  readonly files: { [filename: string]: ArrayBuffer };
+}
+
+self.onmessage = (ev: MessageEvent) => {
+  if (isCompileMessage(ev.data)) {
+    compileModel(ev.data)
+      .then((code) => {
+        self.postMessage(code);
+      })
+      .catch((e) => {
+        self.postMessage({ error: e });
+      });
+  }
+};
+
+async function compileModel(spec: CompileMessage): Promise<string> {
+  const v = Object.entries(spec.files)[0];
+  const source = await parseFile(v[0], v[1]);
+
+  return compileModelFromLexicalModelSource({
```

Comment:
**[Question]**: this return seems like an object to me rather than a Promise with a string type ?